pkg/client/ld/client.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ld
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	ldcmd "github.com/hyperledger/aries-framework-go/pkg/controller/command/ld"
+	ldrest "github.com/hyperledger/aries-framework-go/pkg/controller/rest/ld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/ldcontext"
+)
+
+var logger = log.New("aries-framework/client/ld")
+
+// HTTPClient represents an HTTP client.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Client represents a REST client for JSON-LD API.
+type Client struct {
+	endpoint   string
+	httpClient HTTPClient
+}
+
+// NewClient returns a new JSON-LD REST client.
+func NewClient(endpoint string, opts ...Option) *Client {
```

Comment:
just curious: which `endpoint` is this? remote provider?

Reply:
this is agent controller endpoint, the one for remote provider is passed to `AddRemoteProvider` method as `remoteProviderEndpoint` argument

Reply:
but `pkg/client` is typically SDK clients not the client's for agent's rest controller.

It should be other way around where command controllers use clients for operations.

Reply:
Thanks Sudesh for catching that 👍. I see that this is not in line with what other clients do here, so, for consistency, it would be better to remove that functionality. If SDK client for JSON-LD stuff is something that we really need, we can implement it in a separate task.

Reply:
Yes, because some AFG clients don't need controllers, they can work with pure SDK for all the features.

You already have APIs just call them directly here instead of going through controller.

Reply:
The problem here is that I don't have dedicated service for JSON-LD functionality (e.g. as in case of `wallet`). Command handlers act here as a "domain layer". Probably I will need to create a separate package on a root level (`pkg\ld`), put core logic there and then use it from command handlers and SDK client. I don't think that calling command handlers from the SDK client would be a good idea. What do you think on that? cc @sudeshrshetty

Reply:
better moving core logic to separate package, we did something similar for `pkg/wallet`,  and we have `vcwallet` package for SDK, command controller built on top of `pkg/wallet`